cirq/contrib/paulistring/convert_to_pauli_string_phasors.py

```diff
@@ -83,7 +83,8 @@ def _matrix_to_pauli_string_phasors(self,
             else:
                 pauli_string = ops.PauliString.from_single(qubit, pauli)
                 out_ops.append(
-                    PauliStringPhasor(pauli_string, half_turns=half_turns))
+                    PauliStringPhasor(pauli_string,
+                                      half_turns=round(half_turns, 10)))
```

Comment:
Change the assertion to a `np.testing.assert_allclose`

Reply:
I was hesitant about changing a test. But now with the go-ahead, here we go!

Reply:
Wait, really? And it started failing because of the changes we made? This is very surprising.

It must be that the matrix is slightly different, and the code is using the matrix decomposition. I'll need to think about this for a bit. We might need to add approximate equality to a lot of classes to properly fix this.

Reply:
>It must be that the matrix is slightly different, and the code is using the matrix decomposition

That's the reason for that almost insignificant rounding error? Here are the compared outputs side by side

```
cirq.Circuit(moments=[
    cirq.Moment(operations=[
        PauliStringPhasor({+, 0:Y}, half_turns=-0.5),
    ]),
    cirq.Moment(operations=[
        cirq.CZ.on(cirq.LineQubit(0), cirq.LineQubit(1)),
    ]),
    cirq.Moment(operations=[
        PauliStringPhasor({+, 0:Z}, half_turns=-0.12499999999999978),
    ]),
    cirq.Moment(operations=[
        PauliStringPhasor({+, 0:X}, half_turns=0.5),
    ]),
    cirq.Moment(operations=[
        PauliStringPhasor({+, 0:Z}, half_turns=0.5),
    ]),
])
cirq.Circuit(moments=[
    cirq.Moment(operations=[
        PauliStringPhasor({+, 0:Y}, half_turns=-0.5),
    ]),
    cirq.Moment(operations=[
        cirq.CZ.on(cirq.LineQubit(0), cirq.LineQubit(1)),
    ]),
    cirq.Moment(operations=[
        PauliStringPhasor({+, 0:Z}, half_turns=-0.125),
    ]),
    cirq.Moment(operations=[
        PauliStringPhasor({+, 0:X}, half_turns=0.5),
    ]),
    cirq.Moment(operations=[
        PauliStringPhasor({+, 0:Z}, half_turns=0.5),
    ]),
])
```

Reply:
Well, previously we were making the matrix elements very directly as sqrt(0.5). Now we're doing stuff like adding 1 to sqrt(2) and then multiplying and adding stuff together. It could easily result in being off by an extra ulp, which could propagate through the gate decomposition code and result in an ever so slightly different phase.

Reply:
If you want, you can put the "round" code back but add a message to the issue I opened mentioning that it needs to be removed when the approx circuit equality is added in.